packages/agreement-process/src/services/agreementContractBuilder.ts

```diff
@@ -161,153 +131,55 @@ const getAttributesData = async (
-  }
-
-  if (!producer.selfcareId) {
-    throw agreementSelfcareIdNotFound(producer.id);
-  }
-  if (!consumer.selfcareId) {
-    throw agreementSelfcareIdNotFound(consumer.id);
-  }
-
-  const producerSelfcareId: SelfcareId = unsafeBrandId(producer.selfcareId);
-  const consumerSelfcareId: SelfcareId = unsafeBrandId(consumer.selfcareId);
-
-  /**
-   * The user that activated the agreement, the one in the activation.who stamp, could both belong to the producer institution or the consumer institution.
-   * In case the user is not found with the producer institutionId, we try to retrieve it from the consumer selfcare.
-   */
-
-  // eslint-disable-next-line functional/no-let, @typescript-eslint/no-non-null-assertion
-  let user: selfcareV2ClientApi.UserResponse = undefined!;
-
-  try {
-    user = await retrieveUser(
-      selfcareV2Client,
-      producerSelfcareId,
-      activation.who,
-      correlationId
-    );
-  } catch (e) {
-    if (isAxiosError(e) && e.response?.status === 404) {
-      user = await retrieveUser(
-        selfcareV2Client,
-        consumerSelfcareId,
-        activation.who,
-        correlationId
-      );
-    } else {
-      throw e;
-    }
-  }
-
-  if (user.name && user.surname && user.taxCode) {
-    return [`${user.name} ${user.surname} (${user.taxCode})`, activation.when];
-  }
-
-  throw agreementMissingUserInfo(activation.who);
-};
-
 const getPdfPayload = async (
   agreement: Agreement,
   eservice: EService,
   consumer: Tenant,
   producer: Tenant,
-  readModelService: ReadModelService,
-  selfcareV2Client: SelfcareV2UsersClient,
-  correlationId: CorrelationId
+  readModelService: ReadModelService
 ): Promise<AgreementContractPDFPayload> => {
-  const getTenantText = (name: string, origin: string, value: string): string =>
-    origin === "IPA" ? `${name} (codice IPA: ${value})` : name;
+  const getIpaCode = (tenant: Tenant): string | undefined =>
+    tenant.externalId.origin === PUBLIC_ADMINISTRATIONS_IDENTIFIER
+      ? tenant.externalId.value
+      : undefined;
 
   const today = new Date();
-  const producerText = getTenantText(
-    producer.name,
-    producer.externalId.origin,
-    producer.externalId.value
-  );
-
-  const consumerText = getTenantText(
-    consumer.name,
-    consumer.externalId.origin,
-    consumer.externalId.value
-  );
-  const [submitter, submissionTimestamp] = await getSubmissionInfo(
-    selfcareV2Client,
-    consumer,
-    agreement,
-    correlationId
-  );
-  const [activator, activationTimestamp] = await getActivationInfo(
-    selfcareV2Client,
-    producer,
-    consumer,
-    agreement,
-    correlationId
-  );
 
   const { certified, declared, verified } = await getAttributesData(
     consumer,
     agreement,
     readModelService
   );
 
+  const descriptor = eservice.descriptors.find(
+    (d) => d.id === agreement.descriptorId
+  );
+
+  if (!descriptor) {
+    throw descriptorNotFound(eservice.id, agreement.descriptorId);
+  }
```

Comment:
It could be moved in a separate function getDescriptor similar other getter function.